File: tests/manual_verify.js
const { chromium } = require('playwright-core');

(async () => {
  const browser = await chromium.launch({ executablePath: '/usr/bin/google-chrome' });
  const page = await browser.newPage();
  await page.goto('http://localhost:8000');

  console.log('Testing accessibility attributes...');
  const btn = await page.$('#dashboardBtn');
  const ariaControls = await btn.getAttribute('aria-controls');
  const ariaExpanded = await btn.getAttribute('aria-expanded');
  const title = await btn.getAttribute('title');

  console.log('aria-controls:', ariaControls);
  console.log('aria-expanded:', ariaExpanded);
  console.log('title:', title);

  if (ariaControls !== 'dashboardModal' || ariaExpanded !== 'false' || title !== 'Dashboard (D)') {
    console.error('FAILED accessibility check');
    process.exit(1);
  }

  console.log('Testing keyboard shortcut D to open...');
  await page.keyboard.press('d');
  const ariaExpandedOpen = await btn.getAttribute('aria-expanded');
  console.log('aria-expanded (open):', ariaExpandedOpen);
  if (ariaExpandedOpen !== 'true') {
    console.error('FAILED keyboard open');
    process.exit(1);
  }

  console.log('Testing keyboard shortcut D to close...');
  await page.keyboard.press('d');
  const ariaExpandedClosed = await btn.getAttribute('aria-expanded');
  console.log('aria-expanded (closed):', ariaExpandedClosed);
  if (ariaExpandedClosed !== 'false') {
    console.error('FAILED keyboard close');
    process.exit(1);
  }

  console.log('Testing Escape key to close...');
  await page.keyboard.press('d');
  await page.keyboard.press('Escape');
  const ariaExpandedEsc = await btn.getAttribute('aria-expanded');
  console.log('aria-expanded (esc):', ariaExpandedEsc);
  if (ariaExpandedEsc !== 'false') {
    console.error('FAILED escape close');
    process.exit(1);
  }

  console.log('ALL MANUAL CHECKS PASSED');
  await browser.close();
})();

File: tests/manual_verify.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        page.on("console", lambda msg: print(f"BROWSER CONSOLE: {msg.text}"))

        await page.goto('http://localhost:8000')
        await asyncio.sleep(2)

        btn = page.locator('#dashboardBtn')

        print('Testing click to open...')
        await btn.click()
        await asyncio.sleep(1)
        aria_expanded_open = await btn.get_attribute('aria-expanded')
        print(f'aria-expanded (open) after click: {aria_expanded_open}')

        if aria_expanded_open == 'true':
            await page.screenshot(path="/home/jules/verification/dashboard_open_click.png")
            print("Successfully opened dashboard via click")
        else:
            print("Failed to open dashboard via click")

        print('Testing keyboard shortcut d...')
        # Close it first if open
        if aria_expanded_open == 'true':
            await page.keyboard.press('Escape')
            await asyncio.sleep(1)

        await page.keyboard.press('d')
        await asyncio.sleep(1)
        aria_expanded_d = await btn.get_attribute('aria-expanded')
        print(f'aria-expanded after d: {aria_expanded_d}')

        await browser.close()

if __name__ == '__main__':
    asyncio.run(main())
